File: leetcode/537_Complex_Number_Multiplication/537_Complex_Number_Multiplication.cpp
class Solution {
public:
    string complexNumberMultiply(string a, string b) {
        int a1 = stoi(a.substr(0, a.find('+', 1)));
        int a2 = stoi(b.substr(0, b.find('+', 1)));
        int b1 = stoi(a.substr(a.find('+', 1) + 1, a.find('i') - a.find('+', 1) + 1));
        int b2 = stoi(b.substr(b.find('+', 1) + 1, b.find('i') - b.find('+', 1) + 1));
        return to_string(a1 * a2 - b1 * b2) + "+" + to_string(a1 * b2 + a2 * b1) + "i";
    }
};

File: leetcode/537_Complex_Number_Multiplication/537_Complex_Number_Multiplication.java
class Solution {
    public int findA(String n) {
        StringBuffer sb = new StringBuffer();
        for (int i = 0; i < n.length() && n.charAt(i) != '+'; i++) {
            sb.append(n.charAt(i));
        }
        return Integer.parseInt(sb.toString());
    }
    public int findB(String n) {
        return Integer.parseInt(n.substring(n.indexOf("+", 1) + 1, n.indexOf("i")));
    }
    public String complexNumberMultiply(String a, String b) {
        int a1 = findA(a);
        int a2 = findA(b);
        int b1 = findB(a);
        int b2 = findB(b);
        return (new Integer(a1 * a2 - b1 * b2).toString()) + "+" + (new Integer(a1 * b2 + a2 * b1).toString()) + "i";
    }
}
